RHDEVDOCS-3104: Add docs for editing BuildConfig using Developer perspective (#36660)
openshift/openshift-docs: modules/builds-basic-edit-buildconfig.adoc

// Module included in the following assemblies:
// * builds/basic-build-operations.adoc

[id="builds-basic-edit-buildconfig_{context}"]
= Editing a BuildConfig

> 

Reviewer: What is a `BuildConfig` and why might I want to edit one?

Reviewer: I'll expand on this comment. The content assumes that the reader is familiar with the "BuildConfig" term and its purpose. It almost sounds like jargon.  It would be good to start with an abstract. Maybe something like:

"A your <something> has a default build configuration, also known as a _BuildConfig_. To <give a reason>, you can edit the build configuration."

> You can use the *Edit BuildConfig* option in the *Builds* view of the *Developer* perspective to edit your build configurations. You can edit a `BuildConfig` using the following options:

Reviewer: ```suggestion
To edit your build configurations, you use the *Edit BuildConfig* option in the *Builds* view of the *Developer* perspective . 

You can use either of the following views to edit a `BuildConfig`:
```

> * *Form view*: This allows you to edit and configure your `BuildConfig` using the standard form fields and checkboxes.

Reviewer: ```suggestion
* The *Form view* enables you to edit and configure your `BuildConfig` using the standard form fields and checkboxes.
```
s/allows/enables/ to avoid anthropomorphism.

> * *YAML view*: This allows you to edit and configure your `BuildConfig` manually with full control over the operations.

Reviewer: Same here.

Author: - Thanks for this suggestion. Learnt something new today :smile: 
- The users will not be editing it using the form but have to edit through the raw YAML.
- Yes. Some of the advanced fields are not covered by the form and require the users to edit the YAML.

> You can also switch between the *Form view* and *YAML view* without losing any data. The data in the *Form view* is transferred to the *YAML view* and vice versa.

Reviewer: Good note. 
```suggestion
You can switch between the *Form view* and *YAML view* without losing any data. The data in the *Form view* is transferred to the *YAML view* and vice versa.
```

> Use the following instructions to edit the `BuildConfig` using the Form view of the Developer perspective:

Reviewer: ```suggestion
.Procedure
```

> * In the Builds view of the Developer perspective, click on the menu {kebab} to see *Edit BuildConfig*.

Author: I completely missed this. It should be fixed now.

> * Click on *Edit BuildConfig* to find the *Form view* configure option.

Reviewer: s/click on/click/ throughout.

> * In the Git section, enter the Git repository URL for the codebase you want to use to create an application. The URL is then validated.

Reviewer: Is "Git" a UI element? How do you format UI elements?

> * Optional: You can click *Show Advanced Git Options* to add details such as:

Reviewer: ```suggestion
* Optional: Click *Show Advanced Git Options* to add details such as:
```

> ** *Git Reference* to point to code in a specific branch, tag, or commit to be used to build the application.

Reviewer: ```suggestion
** *Git Reference* to specify a branch, tag, or commit that contains code you want to use to build the application.
```

> ** *Context Dir* to specify the subdirectory for the application source code you want to use to build the application.

Reviewer: ```suggestion
** *Context Dir* to specify the subdirectory that contains code you want to use to build the application.
```

> ** *Source Secret* to create a Secret Name with credentials for pulling your source code from a private repository.

Reviewer: Is "Secret Name" a UI element name?

> * In the *Build from* section, click on the option that you would like to build from. You can use the following options:

Reviewer: If the user can select only one item from a range of choices, you can tell them to "select" or "choose" instead of "click."

> ** *Image Stream tag*: The Image Stream tag is used to reference an image for a given image stream and tag. You will need to enter the project, image stream and tag to refer to where you would like to build from and push to.

Reviewer: ```suggestion
** *Image Stream tag* references an image for a given image stream and tag. Enter the project, image stream, and tag of the location you would like to build from and push to.
```

Reviewer: Oxford comma.

> ** *Image Stream image*: The Image Stream image is used to reference an image for a given image stream and image name. You will need to enter the image stream image from where you would like to build. You will also need to enter the project, image stream and the tag to refer to where you would like to push to.

Reviewer: ```suggestion
** *Image Stream image* references an image for a given image stream and image name. You will need to enter the image stream image from where you would like to build. You will also need to enter the project, image stream and the tag to refer to where you would like to push to.
```
Avoid repetition.
Avoid future tense.
Use imperative verbs to give instructions.

> ** *Docker image*: The Docker image is referenced through a Docker image repository. You will also need to enter the project, image stream and the tag to refer to where you would like to push to.

Reviewer: ```suggestion
** *Docker image*: The Docker image is referenced through a Docker image repository. You will also need to enter the project, image stream, and tag to refer to where you would like to push to.
```

> * In the *Environment Variables* section, you can add all the environment variables associated with the project by adding the *Name* and *Value* field. You can use the *Add Value* option or use the *Add from ConfigMap* and *Secret* to add more environment variables.

Reviewer: Should this step begin with `Optional:`?
```suggestion
* Optional: In the *Environment Variables* section, add the environment variables associated with the project by using the *Name* and *Value* fields. To add more environment variables, use *Add Value*, or use *Add from ConfigMap* and *Secret* .
```

> * Optional: You can use the following advanced options to further customize your application:

Reviewer: ```suggestion
* Optional: To further customize your application, use the following advanced options:
```

Trigger::

> Click on the checkbox to trigger a new image build when the builder image changes. You can add more triggers by clicking on *Add Trigger* option, where you need to select the *Type* and *Secret*.

Reviewer: Can you make similar improvements to the following definitions?

Secrets::
Click on the *Add secret* to add secrets for your application. You can also choose the names to access advanced options where you need to select the *Secret* and *Mount point*.

Policy::
Click on the *Run policy* to select the build run policy. The policy that is selected will further determine the order in which builds created from the build configuration should run.

Hooks::

> Click on the checkbox *Run build hooks after image is built* to run commands at the end of the build to verify the image. You can add the *Hook type*, the *Command* and finally the *Arguments* to be appended to the command.

Reviewer: ```suggestion
To run commands at the end of the build and verify the image, select *Run build hooks after image is built*. You can enter the *Hook type*, *Command*, and *Arguments* to append to the command.
```
